customers.go

```diff
@@ -169,6 +183,35 @@ func (c customerRepository) Update(customer shopify.Customer) (shopify.Customer,
 	return response.Customer.ToShopify(), nil
 }
 
+func (c customerRepository) GetByQuery(fields []string, query shopify.CustomerSearchQuery) (shopify.Customers, error) {
+	url := c.createURL(fmt.Sprintf("customers/search.json?fields=%v&query=%s", strings.Join(fields, ","), query.String()))
+
+	body, _, err := c.client.Get(url, nil)
+	if err != nil {
+		switch err.(http.ErrHTTP).Code {
+		case httpCode.StatusNotFound:
+			return shopify.Customers{}, shopify.NewErrCustomerSearchNotFound(query.String())
```

Comment:
Same here. What does Shopify return if no customer was found?

Reply:
An empty array 😄

Reply:
So not an error :D

Reply:
Nope 😁